certificate_watcher.py

```diff
@@ -69,6 +70,28 @@ def __repr__(self):
         return self.description
 
 
+class CertificateValidationError(Exception):
+    pass
+
+
+def validate_certificate(service: Service, limit: timedelta):
+    try:
+        cert = get_server_certificate(service)
+    except socket.timeout:
+        raise CertificateValidationError(f"{service}: connect timeout")
```

Comment:
```suggestion
    except socket.timeout as err:
        raise CertificateValidationError(f"{service}: connect timeout") from err
```

Else Python will display "During the handling of the error, another error occured", which is misleading. With the raise it'll display something like "The exception is the cause of the other one", which is right.